src/PlusDataCollection/WinProbe/vtkPlusWinProbeVideoSource.cxx

```diff
@@ -1198,41 +1198,91 @@ int32_t vtkPlusWinProbeVideoSource::GetSpatialCompoundCount()
 }
 
 //----------------------------------------------------------------------------
-
-void vtkPlusWinProbeVideoSource::SetBRFEnabled(bool value)
+PlusStatus vtkPlusWinProbeVideoSource::SetExtraSourceMode(Mode mode)
 {
-  if(Connected)
+  if ((GetBRFEnabled() && mode == Mode::BRF) || (GetMModeEnabled() && mode == Mode::M) || (GetARFIEnabled() && mode == Mode::ARFI))
   {
-    if(m_Mode == Mode::M)
-    {
-      SetMIsEnabled(false);
-    }
-    if(m_Mode == Mode::ARFI)
-    {
-      SetARFIEnabled(false);
-    }
-    if(value)
-    {
-      SetHandleBRFInternally(false);
-      SetBFRFImageCaptureMode(2);
-    }
-    else
-    {
-      SetHandleBRFInternally(true);
-      SetBFRFImageCaptureMode(0);
-    }
+    return PLUS_SUCCESS;
   }
+  SetMIsEnabled(mode == Mode::M);
+  SetARFIIsEnabled(mode == Mode::ARFI);
 
-  if(value)
+  if (mode == Mode::BRF)
   {
-    m_Mode = Mode::BRF;
+    SetHandleBRFInternally(false);
+    SetBFRFImageCaptureMode(2);
   }
   else
   {
-    m_Mode = Mode::B;
+    SetHandleBRFInternally(true);
+    SetBFRFImageCaptureMode(0);
+  }
+
+  if(mode == Mode::M)
+  {
+    SetMIsRevolving(m_MRevolvingEnabled);
+    SetMPRF(m_MPRF);
+    SetMAcousticLineIndex(m_MLineIndex);
+    ::SetMWidth(2000);
+    // ::SetMWidth(m_ExtraFrameSize[0]);
```

Comment:
Hardcoding the value helps when going from RF to MMode and having the correct X dimension value. It also helps correctly go from `SetARFIEnabled` to `SetMModeEnabled` and have MMode live streaming again.

We should figure out how not to hardcode here.